doc/docs/Python_Tutorials/Custom_Source.md

```diff
@@ -375,3 +379,228 @@ $$f_{m,n}(x,y) = \sqrt{\frac{c_m c_n}{L_x L_y}} \cos \left(\frac{m\pi x}{L_x}\ri
+
+    sim = mp.Simulation(
+        cell_size=cell_size,
+        resolution=resolution,
+        k_point=mp.Vector3(),
+        boundary_layers=pml_layers,
+        geometry=geometry,
+        sources=sources,
+    )
+
+    dft_mons = []
+    for nd in range(ndipole):
+        dft_mons.append(
+            sim.add_dft_fields(
+                [mp.Ez],
+                fcen,
+                df,
+                nfreq,
+                center=mp.Vector3(
+                    sx*(-0.5 + nd/ndipole),
+                    -0.5*sy+dAg+0.5*dsub
+                ),
+                size=mp.Vector3(),
+            )
+        )
+
+    run_time = rt * nfreq / df
+    sim.run(until_after_sources=run_time)
+
+    freqs = mp.get_flux_freqs(dft_mons[0])
+
+    abs_flux = np.zeros(nfreq)
+    dft_ez = np.zeros(ndipole,dtype=np.complex128)
+    for nf in range(nfreq):
+        for nd in range(ndipole):
+            dft_ez[nd] = sim.get_dft_array(dft_mons[nd], mp.Ez, nf)
+        abs_flux[nf] = np.sum(np.abs(dft_ez)**2)
+
+    return freqs, abs_flux
+
+
+if __name__ == '__main__':
+    fwd_flat_flux = np.zeros((nfreq,ndipole))
+    fwd_text_flux = np.zeros((nfreq,ndipole))
+    for d in range(ndipole):
+        fwd_freqs, fwd_flat_flux[:,d] = forward(d, 2, False)
+        _, fwd_text_flux[:,d] = forward(d, 4, True)
+
+    fwd_norm_flux = np.mean(fwd_text_flux,axis=1) / np.mean(fwd_flat_flux,axis=1)
+
+    bwd_freqs, bwd_flat_flux = backward(2, False)
+    _, bwd_text_flux = backward(4, True)
+    bwd_norm_flux = bwd_text_flux / bwd_flat_flux
+
+    plt.figure()
+    plt.semilogy(fwd_freqs, fwd_norm_flux, 'b-', label='forward')
+    plt.semilogy(bwd_freqs, bwd_norm_flux, 'r-', label='backward')
+    plt.xlabel("frequency")
+    plt.ylabel("normalized flux")
+    plt.legend()
+
+    if mp.am_master():
+        plt.savefig(
+            "forward_vs_backward_flux_spectrum.png",
+            bbox_inches='tight',
+            dpi=150
+        )
+```
+
+There are four items to note in this script. (1) Since the exact equivalence between the forward and reciprocal calculations only applies to the continuum model, demonstrating agreement using a discretized model typically requires going to fine grid resolutions. In this example, we found that a resolution of 200 pixels/μm produced results which are sufficiently converged. (2) Measuring the flux in the normal direction in the forward calculation requires [mode decomposition](../Python_User_Interface.md#mode-decomposition) rather than a Poynting flux monitor. (3) To compare the results of the forward and reciprocal calculations, similar to the previous examples, the flux spectrum of the grating structure must be normalized using the flux spectrum of a flat structure. (4) Measuring the emission in the normal direction requires specifying a `k_point` of zero. (For emission at angle $\theta$ from the $+y$ axis, the `k_point` would be $\omega(\sin(\theta),\cos(\theta))$ where $\omega$ is the source/monitor frequency.)
```

Comment:
```suggestion
There are four items to note in this script. (1) Since the exact equivalence between the forward and reciprocal calculations only applies to the continuum model, demonstrating agreement using a discretized model typically requires going to fine grid resolutions.  (In principle, there is an exact discrete version of reciprocity, but that would require more precise attention to how the sources and monitors are discretized.) In this example, we found that a resolution of 200 pixels/μm produced which match between forward and reciprocal calculations (a high resolution is required to accurately resolve sharp resonant effects). (2) Measuring the flux in the normal direction in the forward calculation requires [mode decomposition](../Python_User_Interface.md#mode-decomposition) rather than a Poynting flux monitor. (3) To compare the results of the forward and reciprocal calculations, similar to the previous examples, the flux spectrum of the grating structure must be normalized using the flux spectrum of a flat structure. (4) Measuring the emission in the normal direction requires specifying a `k_point` of zero. (For emission at angle $\theta$ from the $+y$ axis, the `k_point` would be $\omega(\sin(\theta),\cos(\theta))$ where $\omega$ is the source/monitor frequency.)
```